File: files/100-199/195/CH-195.py
import pandas as pd

path = "CH-195 Missing Char.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=6).rename(columns=lambda x: x.split('.')[0])

def transform_text(text):
    chars = text[:2] 
    
    for i in range(2, len(text)):
        char = text[i] 
        cond = ((len(chars) + 1) % 3 == 0) and (char != "/")  
        chars += ('-' if cond else '') + char 
    
    return chars

input['result'] = input.iloc[:, 0].apply(transform_text)

result = input['result'].tolist()
expected = test.iloc[:, 0].tolist()

print(result == expected) # True

File: files/100-199/195/CH-195.R
library(tidyverse)
library(readxl)

path = "files/CH-195 Missing Char.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:D7")

transform_text <- function(text) {
  chars <- substr(text, 1, 2) 
  
  for (i in seq(3, nchar(text))) {
    char <- substr(text, i, i)  
    cond <- ((nchar(chars) + 1) %% 3 == 0) && (char != "/") 
    chars <- paste0(chars, ifelse(cond, "-", ""), char)
  }
  
  return(chars)
}

result = input %>%
  mutate(result = map_chr(ID, transform_text))

all.equal(result$result, test$ID, check.attributes = FALSE) # TRUE
